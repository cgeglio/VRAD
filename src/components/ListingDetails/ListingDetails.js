import React from 'react';
import PropTypes from 'prop-types';
import './ListingDetails.scss';
import 'react-responsive-carousel/lib/styles/carousel.min.css';
import { Carousel } from 'react-responsive-carousel';

const ListingDetails = ({ name, listing_id, address, details, dev_id }) => {
  const createList = details.features.map((feature, i) => <li key={dev_id + [i]}>{feature}</li>);

  return (
    <article className="listing">
      <Carousel className="carousel" width="700px">
        <div>
          <img className="listing-img" src={process.env.PUBLIC_URL + `/images/${listing_id}_a.jpg`} alt="listing 1" />
        </div>
        <div>
          <img className="listing-img" src={process.env.PUBLIC_URL + `/images/${listing_id}_b.jpg`} alt="listing 2" />
        </div>
        <div>
          <img className="listing-img" src={process.env.PUBLIC_URL + `/images/${listing_id}_c.jpg`} alt="listing 3" />
        </div>
      </Carousel>

      <section className="listing-details">
        <div className="title">
          <h2>{name}</h2>
          <p>{address.street}, {address.zip}</p>
        </div>
        <div>
          <p className="cost">${details.cost_per_night} per night</p>
        </div>
        <div className="bed-bath">
          <p>Beds: {details.beds} / Baths: {details.baths}</p>
        </div>
        <div className="features">
          <h3>Features:</h3>
          <ul>{createList}</ul>
        </div>
      </section>
    </article>
  );
};

export default ListingDetails;

ListingDetails.propTypes = {
  name: PropTypes.string.isRequired,
  listing_id: PropTypes.number.isRequired,
  address: PropTypes.object.isRequired,
  details: PropTypes.object.isRequired,
};
