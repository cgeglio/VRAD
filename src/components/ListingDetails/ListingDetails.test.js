import React from 'react';
import { shallow } from 'enzyme';
import ListingDetails from './ListingDetails';

describe('ListingDetails', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const mockListing = {
      listing_id: 5,
      name: 'test name',
      address: { street: 'test street' },
      details: { features: ['test feature'] },
    };
    const wrapper = shallow(<ListingDetails
      key={mockListing.key}
      {...mockListing}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
