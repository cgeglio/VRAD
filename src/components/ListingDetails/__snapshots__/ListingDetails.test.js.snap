// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingDetails should match the snapshot with all data passed in correctly 1`] = `
<article
  className="listing"
>
  <Carousel
    autoPlay={false}
    axis="horizontal"
    centerMode={false}
    centerSlidePercentage={80}
    className="carousel"
    dynamicHeight={false}
    emulateTouch={false}
    infiniteLoop={false}
    interval={3000}
    labels={
      Object {
        "item": "slide item",
        "leftArrow": "previous slide / item",
        "rightArrow": "next slide / item",
      }
    }
    onChange={[Function]}
    onClickItem={[Function]}
    onClickThumb={[Function]}
    onSwipeEnd={[Function]}
    onSwipeMove={[Function]}
    onSwipeStart={[Function]}
    selectedItem={0}
    showArrows={true}
    showIndicators={true}
    showStatus={true}
    showThumbs={true}
    statusFormatter={[Function]}
    stopOnHover={true}
    swipeScrollTolerance={5}
    swipeable={true}
    transitionTime={350}
    useKeyboardArrows={false}
    verticalSwipe="standard"
    width="700px"
  >
    <div>
      <img
        alt="listing 1"
        className="listing-img"
        src="/images/5_a.jpg"
      />
    </div>
    <div>
      <img
        alt="listing 2"
        className="listing-img"
        src="/images/5_b.jpg"
      />
    </div>
    <div>
      <img
        alt="listing 3"
        className="listing-img"
        src="/images/5_c.jpg"
      />
    </div>
  </Carousel>
  <section
    className="listing-details"
  >
    <div
      className="title"
    >
      <h2>
        test name
      </h2>
      <p>
        test street
        , 
      </p>
    </div>
    <div>
      <p
        className="cost"
      >
        $
         per night
      </p>
    </div>
    <div
      className="bed-bath"
    >
      <p>
        Beds: 
         / Baths: 
      </p>
    </div>
    <div
      className="features"
    >
      <h3>
        Features:
      </h3>
      <ul>
        <li
          key="0.7972115091313292"
        >
          test feature
        </li>
      </ul>
    </div>
  </section>
</article>
`;
